### PR TITLE
Backend feature/update models

### LINKED ISSUE #197: Divide client Model into two models named customer and vendor
**Describe the feature**
We will divide client model into two models named customer and vendor. This way the distinction between them will be more clear in the code.

**Describe solutions you've considered**
We will change all related files about client model, this will affect nearly all the files in the code base.

**Additional context**

### PR DESCRIPTION
**Pull request by mtyazici**
Things done for this PR:
- I added two new models named customer and vendor. They inherit from client model. 
- I put the common endpoints for customer and vendor to a common router named `client`. 
- I changed the controllers, services and routers so that vendor and customer models could be accessed via our API.
- I deleted Google functionality from vendor model.

Things to be done in future:
- We should create the necessary models for fields in vendor and customer model. Right now their type are all strings.
- I updated the api-docs but they could be improved.
- We could reduce signupWithGoogle and loginWithGoogle to only SigninWithGoogle

### CODE REVIEW

File: backend/models/vendor.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const mongoose = require("mongoose");
+const ClientBase = require("./client");
+
+const Schema = mongoose.Schema;
+
+var customerSchema = new Schema(
```
Comment (oztasoi) [inline backend/models/vendor.js]: @mtyazici var vendorSchema = new Schema(

File: backend/models/vendor.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    companyDomainName: { type: String },
+    aboutCompany: { type: String },
+    IBAN: { type: String },
+    addresses: { type: String },
+    location: { type: String },
+  },
+  { collection: "Clients" }
+);
+
+ClientBase.discriminator("Vendor", customerSchema);
```
Comment (oztasoi) [inline backend/models/vendor.js]: ClientBase.discriminator("Vendor", vendorSchema);

**Review by oztasoi — CHANGES_REQUESTED** [review]
@mtyazici I've commented necessary changes, sorry not to change on-review. 😁

File: backend/models/vendor.js (hunk 1 above)
Comment (mtyazici) [reply backend/models/vendor.js]: updated :+1:

File: backend/models/vendor.js (hunk 2 above)
Comment (mtyazici) [reply backend/models/vendor.js]: updated :+1:

**Comment by mtyazici** [comment]
> @mtyazici I've commented necessary changes, sorry not to change on-review. grin

That's fine :smile:  But I think it is better to show changes here so we can commit them right on the website.

**Review by oztasoi — APPROVED** [review]
@mtyazici now it seems just fine 💯  way to go 🚀

**Pull request merged**